tests/lean/run/slice.lean

```diff
@@ -0,0 +1,29 @@
+import Std.Data.Iterators
```

Comment:
Just out of curiosity, how many of these are solved by `simp` today?

Reply:
None so far: Lemmas haven't been a priority yet, but I plan to add some in a later PR.